lib/signs/bus.ex

```diff
@@ -434,34 +455,57 @@ defmodule Signs.Bus do
          state
        ) do
     %{top_configs: top_configs, bottom_configs: bottom_configs} = state
+    stop_ids = all_stop_ids(state)
     top_content = configs_content(top_configs, predictions_lookup, route_alerts_lookup)
     bottom_content = configs_content(bottom_configs, predictions_lookup, route_alerts_lookup)
 
     if !Enum.any?(top_content ++ bottom_content, &match?({:predictions, _}, &1)) &&
-         all_stop_ids(state)
+         stop_ids
          |> Enum.all?(fn stop_id ->
            Map.get(stop_alerts_lookup, stop_id) in @alert_levels
          end) do
       no_service_content()
     else
+      silver_line_headway =
+        combine_silver_line_headway_messages(
+          silver_line_headway_message(top_configs, stop_ids, current_time),
+          silver_line_headway_message(bottom_configs, stop_ids, current_time)
+        )
+
       messages =
-        for content <- [top_content, bottom_content] do
-          Enum.map(content, &format_message(&1, nil, current_time))
-          |> paginate_message()
+        if top_content ++ bottom_content == [] do
+          format_message(silver_line_headway, current_time)
+        else
+          for {content, config} <- [{top_content, top_configs}, {bottom_content, bottom_configs}] do
+            if content == [] do
+              headway_message_for_configs(config, stop_ids, current_time)
+              |> format_message(current_time)
```

Comment:
I don't think this will handle the case where one direction has predictions and the other is showing headways. We should have a test case for that anyway, so writing that will be a good starting point. `format_message` is expected to produce single-line content, so we should keep it that way and have it call `Message.to_single_line` for headways. We'll also need to port/borrow some of the sizing and flipping logic from `Utilities.Messages.render_messages`, because single-line headway messages are too long for the top line.